clang/test/CXX/drs/dr28xx.cpp

```diff
@@ -58,3 +58,24 @@ void B<int>::g() requires true;
 #endif
 
 } // namespace dr2847
+
+namespace dr2858 { // dr2858: 19
+
+#if __cplusplus > 202302L
+
+template<typename... Ts>
+struct A {
+  // FIXME: The nested-name-specifier in the following friend declarations are declarative,
+  // but we don't treat them as such (yet).
+  friend void Ts...[0]::f();
+  template<typename U>
+  friend void Ts...[0]::g();
+
+  friend struct Ts...[0]::B;
+  template<typename U>
+  friend struct Ts...[0]::C; // expected-warning{{is not supported; ignoring this friend declaration}}
```

Comment:
Like the rest of the file, you should use `expected-warning@-1` style, and include full text of the diagnostic you're matching.

Reply:
@Endilll The index of the _pack-index-specifier_ is currently printed as a memory address (as opposed to `0`). I'll be opening another patch to fix that, but that is why some of the diagnostic message is omitted.

Reply:
Oh, that's unfortunate. You can still use `expected-warning-re@-1`, and omit the address via regular expression.